src/critnib/critnib.c

```diff
@@ -176,37 +182,68 @@ static inline unsigned slice_index(word key, sh_t shift) {
  * critnib_new -- allocates a new critnib structure
  */
 struct critnib *critnib_new(void) {
-    struct critnib *c = Zalloc(sizeof(struct critnib));
-    if (!c) {
+    umf_ba_linear_pool_t *pool_linear =
+        umf_ba_linear_create(0 /* minimal pool size */);
```

Comment:
shouldn't we explicitly request a proper size? It's probably unlikely but it may happen that those metada will take more than 4K in future

Reply:
We allocate only `sizeof(struct critnib) = 352` bytes now, so 4K is more than enough now. We can increase that if we need more than 4K at any moment.

Reply:
But will you remember to do that? ;d The 4k is not documented anywhere so perhaps we should at least add some comment / assert for the size as a reminder?

Reply:
It will start failing then.

Reply:
Ok, I forgot we don't have any builtin static_assert in C so we can only check it at runtime anyway.